doc/design/staging-1.md

```diff
@@ -0,0 +1,37 @@
+# Staging Environment and Procedures
+
+## Production Environment
+* Django API server, Postgres, and CloudAMQP running in Heroku
+* Sponsored S3 bucket in the sponsored account in AWS
+* Web GUI served through Netlify
+* Sentry instance
+
+## Staging Environment
+* Django API server, Postgres, and CloudAMQP running in Heroku
+* Private S3 bucket in the project account in AWS
+* Web GUI served through Netlify
+* Sentry instance
+
+This should exactly parallel production, except for the S3 bucket, which is private.
+Nothing is actually shared, everything is hosted adjacently.
+
+The staging server will contain some test data and some (smallish) real data that reflects the contents of production.
+
+## Release procedure
+The staging server will track the `master` branch in `dandi-api` and will automatically redeploy whenever `master` is changed.
+
+The production server will not track any branch. Instead, it will automatically redeploy whenever a release is tagged in GitHub.
+
+The staging web GUI will track the `staging` branch in `dandiarchive` and will automatically redeploy whenever `staging` is changed.
```

Comment:
why not to make it consistent between the two (dandi-api and webui) one way or another? 
E.g. the easiest might be to hav `staging` branch in both and keep `master` for auto-deployment (but be careful to not target development PRs against it).  `releases` could be "minted" on top of master at will and/or for each PR against `master`.  "place staging into production" could be a PR to have `staging` merged into `master`.

Alternative naming -- `master` (staging) + `maint` (or `releases`) branch.  One way or another -- consistency would only benefit IMHO.

Reply:
I would like to have the deploy mechanism consistent and just tag a release in both, but the issue is that Netlify only supports automatic deploys from a branch. 

How about having development for `dandiarchive` happen in `master` and have staging track `master` as you suggest, then having a GitHub action that merges `master` into a `releases` branch whenever a release is tagged?